packages/libs/src/sdk/services/Solana/programs/PaymentRouterClient/PaymentRouterClient.ts

```diff
@@ -0,0 +1,251 @@
+import { PaymentRouterProgram } from '@audius/spl'
+import {
+  Account,
+  TokenAccountNotFoundError,
+  TokenInvalidMintError,
+  TokenInvalidOwnerError,
+  createAssociatedTokenAccountIdempotentInstruction,
+  createTransferCheckedInstruction,
+  getAccount,
+  getAssociatedTokenAddressSync
+} from '@solana/spl-token'
+import {
+  PublicKey,
+  TransactionInstruction,
+  TransactionMessage,
+  VersionedTransaction
+} from '@solana/web3.js'
+
+import { productionConfig } from '../../../../config/production'
+import { mergeConfigWithDefaults } from '../../../../utils/mergeConfigs'
+import { mintFixedDecimalMap } from '../../../../utils/mintFixedDecimalMap'
+import { parseParams } from '../../../../utils/parseParams'
+import { Prettify } from '../../../../utils/prettify'
+import { Mint } from '../../types'
+import { BaseSolanaProgram } from '../BaseSolanaProgram'
+
+import { getDefaultPaymentRouterClientConfig } from './getDefaultConfig'
+import {
+  CreateMemoInstructionRequest,
+  CreateMemoInstructionSchema,
+  CreatePurchaseContentInstructionsRequest,
+  CreatePurchaseContentInstructionsSchema,
+  CreateRouteInstructionRequest,
+  CreateRouteInstructionSchema,
+  CreateTransferInstructionRequest,
+  CreateTransferInstructionSchema,
+  GetOrCreateProgramTokenAccountRequest,
+  GetOrCreateProgramTokenAccountSchema,
+  PaymentRouterClientConfig
+} from './types'
+
+export class PaymentRouterClient extends BaseSolanaProgram {
+  private readonly programId: PublicKey
+
+  /** The intermediate account where funds are sent to and routed from. */
+  private readonly programAccount: PublicKey
+  private readonly programAccountBumpSeed: number
+
+  private readonly mints: Prettify<Partial<Record<Mint, PublicKey>>>
+
+  private existingTokenAccounts: Prettify<Partial<Record<Mint, Account>>>
+
+  constructor(config: PaymentRouterClientConfig) {
+    const configWithDefaults = mergeConfigWithDefaults(
+      config,
+      getDefaultPaymentRouterClientConfig(productionConfig)
+    )
+    super(configWithDefaults, config.solanaWalletAdapter)
+    this.programId = configWithDefaults.programId
+    const [pda, bump] = PublicKey.findProgramAddressSync(
+      [new TextEncoder().encode('payment_router')],
+      this.programId
+    )
+    this.programAccount = pda
+    this.programAccountBumpSeed = bump
+    this.mints = configWithDefaults.mints
+    this.existingTokenAccounts = {}
+  }
+
+  public async createTransferInstruction(
+    params: CreateTransferInstructionRequest
+  ) {
+    const args = await parseParams(
+      'crateTransferInstruction',
+      CreateTransferInstructionSchema
+    )(params)
+    const mint = this.mints[args.mint]
```

Comment:
I think it's interesting that some of our utilities/program clients default to a mint and others don't?

Reply:
none of the clients should? where are you seeing that?

Reply:
Let me rephrase 😅 
It looks like _this_ client expects you to pass a mint and doesn't default to something. But I may have missed where it's default-initialized (and I would assume it defaults to usdc)

Reply:
I don't think it's desirable for the clients to default to a mint, personally. The apis should be explicit imo

Reply:
Ah okay. I had thought I saw in other clients we defined here that they _did_ default (Maybe claimable tokens defaulting to waudio?). But I agree with you that they shouldn't, and if I'm imagining things then great!